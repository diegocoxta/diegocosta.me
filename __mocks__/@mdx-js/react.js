module.exports = {
  MDXProvider: jest.fn(({ children }) => children),
};
