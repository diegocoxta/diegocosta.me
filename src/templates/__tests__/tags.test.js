import React from 'react';
import { render } from '@testing-library/react';

jest.mock('../../components/Header', () => () => <p>Header</p>);
jest.mock('../../components/Footer', () => () => <p>Footer</p>);
jest.mock('../../components/ArticleHeader', () => () => <p>ArticleHeader</p>);
jest.mock('../../components/Metatags', () => () => <p>Metatags</p>);

import TagsPage from '../tags';

describe('<TagsPage>', () => {
  it('should render properly', () => {
    const data = {
      articles: {
        totalCount: 2,
        edges: [
          {
            node: {
              frontmatter: {
                title: 'Title 1',
                date: '30/10/2020',
                tags: [],
                description: 'Awesome first article',
              },
              fields: {
                slug: '/path-to-article',
                readingTime: {
                  minutes: 5,
                },
              },
              excerpt: 'Awesome first article',
            },
          },
          {
            node: {
              frontmatter: {
                title: 'Title 2',
                date: '30/10/2020',
                tags: [],
                description: 'Awesome second article',
              },
              fields: {
                slug: '/path-to-article-2',
                readingTime: {
                  minutes: 2,
                },
              },
              excerpt: 'Awesome first article',
            },
          },
        ],
      },
    };

    const { baseElement, getAllByTestId, getByText } = render(
      <TagsPage data={data} pageContext={{ tag: 'Tech Stuff' }} />
    );
    expect(getAllByTestId('tags-page-article').length).toEqual(data.articles.totalCount);
    expect(getByText('Awesome second article')).toBeTruthy();
    expect(getByText('Awesome first article')).toBeTruthy();
    expect(getByText(`Tech Stuff (${data.articles.totalCount})`)).toBeTruthy();
    expect(baseElement).toMatchSnapshot();
  });
});
