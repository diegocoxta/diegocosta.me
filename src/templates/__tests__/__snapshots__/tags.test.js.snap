// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TagsTemplate> renders properly 1`] = `
.c0 {
  max-width: 800px;
  margin: 0 auto;
  padding: 0 20px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 40px 0 0 0;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.c2 {
  margin: 0 5px;
  color: #2F3D4F;
  font-weight: 700;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 20px;
}

.c3 {
  background: linear-gradient(90deg,#ffffff 20px,transparent 1%) center, linear-gradient(#ffffff 20px,transparent 1%) center, #2F3D4F;
  background-size: 22px 22px;
  height: 120px;
  margin: 40px 0;
  width: 100%;
}

.c7 {
  color: #2F3D4F;
  box-shadow: none;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 36px;
  margin: 0;
  font-weight: bold;
}

.c7:hover,
.c7:focus {
  border-bottom: 1px solid #2F3D4F;
  outline: none;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  padding: 0;
  margin: 0;
}

.c5 {
  margin: 0 0 60px 0;
  font-weight: 400;
  color: #2F3D4F;
}

.c6 {
  margin: 20px 0 0 0;
}

.c8 {
  color: #38485D;
  font-size: 18px;
  margin: 5px 0;
}

.c10 {
  line-height: 1.5;
}

.c10 a {
  color: #AAC13D;
  -webkit-text-decoration: none;
  text-decoration: none;
  border-bottom: 1px solid #ffffff;
}

.c10 a:hover,
.c10 a:focus {
  border-bottom: 1px solid #AAC13D;
  outline: none;
}

.c10 blockquote {
  border-left: 5px solid #AAC13D;
  padding-left: 20px;
}

.c10 img {
  box-shadow: none !important;
}

.c10 h3 {
  font-size: 28px;
}

.c4 {
  display: inline-block;
  background: #AAC13D;
  padding: 2px 10px;
  margin: 0px 10px 30px 0;
  color: #fff;
  border-radius: 5px;
  font-size: 48px;
}

<body>
  <div>
    <section
      class="c0"
    >
      <main>
        <div
          class="c1"
        >
          <a
            aria-label="languageswitcher.toggle languages.pt"
            class="c2"
            href="/about-us"
            language="pt"
          >
            /
            pt
          </a>
          <a
            aria-label="languageswitcher.toggle languages.en"
            class="c2"
            href="/about-us"
            language="en"
          >
            en
          </a>
          <a
            aria-label="languageswitcher.toggle languages.es"
            class="c2"
            href="/about-us"
            language="es"
          >
            es
          </a>
        </div>
      </main>
    </section>
    <p>
      Header
    </p>
    <main>
      <p>
        Metatags
      </p>
      <div
        class="c3"
      />
      <section
        class="c0"
      >
        <main>
          <h1
            class="c4"
          >
            Tech Stuff
          </h1>
        </main>
      </section>
      <section
        class="c0"
      >
        <main>
          <article
            class="c5"
            data-testid="article-item"
          >
            <header
              class="c6"
            >
              <a
                class="c7"
                data-testid="article-header-title"
                href="/path-to-article"
              >
                Title 1
              </a>
              <p
                class="c8"
              >
                Invalid Date
                 
                · 5 article.minutes article.ofReading
                 
              </p>
              <ul
                class="c9"
                data-testid="article-header-tags"
              />
            </header>
            <section
              class="c10"
            >
              Awesome first article
            </section>
          </article>
        </main>
      </section>
      <section
        class="c0"
      >
        <main>
          <article
            class="c5"
            data-testid="article-item"
          >
            <header
              class="c6"
            >
              <a
                class="c7"
                data-testid="article-header-title"
                href="/path-to-article-2"
              >
                Title 2
              </a>
              <p
                class="c8"
              >
                Invalid Date
                 
                · 2 article.minutes article.ofReading
                 
              </p>
              <ul
                class="c9"
                data-testid="article-header-tags"
              />
            </header>
            <section
              class="c10"
            >
              Awesome second article
            </section>
          </article>
        </main>
      </section>
    </main>
    <p>
      Footer
    </p>
  </div>
</body>
`;
